packages/bitbucket-eslint-report/index.ts

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env node
+
+import fs from 'fs/promises';
+import path from 'path';
+import commandLineArgs from 'command-line-args';
+import {
+  uploadReportToBitbucket,
+  uploadAnnotationsToBitbucket,
+  createBuildOnBitbucket,
+} from '@zooshdigital/bitbucket-code-insights';
+import { BitbucketAnnotation, BitbucketSeverity } from '@zooshdigital/bitbucket-code-insights/dist/types';
+
+import { createLogger } from 'utils/logger';
+
+type LintMessage = {
+  ruleId: string;
+  severity: number;
+  line: number;
+  column: number;
+  fix?: {
+    text: string;
+  };
+};
+
+type LintFile = {
+  filePath: string;
+  errorCount: number;
+  fixableErrorCount: number;
+  warningCount: number;
+  fixableWarningCount: number;
+  messages: LintMessage[];
+};
+
+const optionDefinitions = [
+  { name: 'name', alias: 'n', type: String },
+  { name: 'path', alias: 'p', type: String, multiple: true },
+  { name: 'add-build', alias: 'a', type: Boolean },
+  { name: 'strict', alias: 's', type: Boolean },
+];
+
+const args = commandLineArgs(optionDefinitions);
+
+const logger = createLogger('Zoosh Bitbucket Eslint Report');
+
+const severities: Record<number, BitbucketSeverity> = {
+  0: 'LOW',
+  1: 'MEDIUM',
+  2: 'HIGH',
+};
+
+async function uploadReport() {
+  const items: BitbucketAnnotation[] = [];
+
+  let counter = 0;
+  let totalErrors = 0;
+  let fixableErrors = 0;
+  let totalWarnings = 0;
+  let fixableWarnings = 0;
+
+  try {
+    const { name, path: reportPaths, 'add-build': addBuild, strict } = args;
+
+    if (!name || !reportPaths || reportPaths.length === 0) {
+      throw new Error(
+        'Bitbucket Eslint Report - Usage: bitbucket-eslint-report -n <report-name> -p <report-path> [-b] [-s]',
+      );
+    }
+
+    await Promise.all(
+      reportPaths.map(async (reportPath: string) => {
+        const lint: LintFile[] = JSON.parse(await fs.readFile(reportPath, 'utf8'));
```

Comment:
Do we need to ensure the JSON follows the expected schema, or it is not our responsibility?

Reply:
This is in a try catch, so schema issue would result in an error caught and reported. I think that's good enough.

Reply:
yes, you're right